src/test/rustdoc-gui/font-weight.goml

```diff
@@ -13,6 +13,10 @@ goto: file://|DOC_PATH|/test_docs/struct.Foo.html
 assert-css: (".impl-items .method", {"font-weight": "600"}, ALL)
 
 goto: file://|DOC_PATH|/lib2/trait.Trait.html
+// This uses '/parent::*' as a proxy for the style of the text node.
```

Comment:
I still it would be much simpler to add a `rustdoc` test checking that there is no `<b>` inside `.constant`.

Something like:

```
@has - 'a[@class="constant"]' 'content'
@!has - 'a[@class="constant"]/b'
```

Reply:
I could do that, but it seems relatively unlikely to me that someone would accidentally add a `<b>` tag. It's much more frequent that rustdoc styling bugs are caused by inadvertent CSS changes, which wouldn't be caught by an `@has` check.

Reply:
For context, the `<b>` tag that I removed had been there for 4 1/2 years (since 081336e8eb5d94071dad7f6fee9efd00305dc381)—when it was added, it worked correctly, but rustdoc's HTML and CSS has since changed such that it is now rendered incorrectly.

Reply:
And also, worst comes to worse, having this regress does not significantly impact the docs. It would just look less consistent.

Reply:
This a good argument. Then please simplify the XPath, I have literally no idea what it's doing currently. XD

Reply:
Which parts would you like me to simplify?

I can also explain what each part means:

- `//*[@class='docblock type-decl']` — selects element of any tag with classes `docblock` and `type-decl`
- `/pre[@class='rust trait']` — selects immediate child with tag `pre` and classes `rust` and `trait`
- `/code` — selects immediate child with tag `code`
- `/a[@class='constant']` — selects immediate child with tag `a` and class `constant`
- `//text()` — selects child that is text node
- `/parent::*` — selects immediate parent of the text node (the `*` means it can be any tag)

Reply:
Please add this whole explanation as a comment then. Once done, r=me.

Reply:
I added a link to my explanation comment on GitHub above because I felt that it would be hard to format the explanation well in a code comment.

Reply:
Sorry but not a big fan of adding a link here. What do you think about:

```
// This is a complex selector, so here how it works:
//
// * //*[@class='docblock type-decl'] — selects element of any tag with classes docblock and type-decl
// * /pre[@class='rust trait'] — selects immediate child with tag pre and classes rust and trait
// * /code — selects immediate child with tag code
// * /a[@class='constant'] — selects immediate child with tag a and class constant
// * //text() — selects child that is text node
// * /parent::* — selects immediate parent of the text node (the * means it can be any tag)
```


?

Reply:
The problem with adding a comment is that it can fall out of date quickly without it being obvious that it's out of date. Whereas if you link to a PR comment, people can easily see that it's old. But, I'll add the comment since I agree that there is some advantage to having it inline :)

Reply:
Thanks!